packages/cli/generators/relation/has-one-relation.generator.js

```diff
@@ -0,0 +1,200 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/cli
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+
+const path = require('path');
+const BaseRelationGenerator = require('./base-relation.generator');
+const relationUtils = require('./utils.generator');
+const utils = require('../../lib/utils');
+
+const CONTROLLER_TEMPLATE_PATH_HAS_ONE =
+  'controller-relation-template-has-one.ts.ejs';
+
+module.exports = class HasOneRelationGenerator extends BaseRelationGenerator {
+  constructor(args, opts) {
+    super(args, opts);
+  }
+
+  async generateControllers(options) {
+    this.artifactInfo.sourceModelClassName = options.sourceModel;
+    this.artifactInfo.targetModelClassName = options.destinationModel;
+    this.artifactInfo.sourceRepositoryClassName =
+      this.artifactInfo.sourceModelClassName + 'Repository';
+    this.artifactInfo.controllerClassName =
+      this.artifactInfo.sourceModelClassName +
+      this.artifactInfo.targetModelClassName +
+      'Controller';
+    this.artifactInfo.paramSourceRepository = utils.camelCase(
+      this.artifactInfo.sourceModelClassName + 'Repository',
+    );
+
+    this.artifactInfo.sourceModelName = utils.toFileName(options.sourceModel);
+    this.artifactInfo.sourceModelPath = utils.pluralize(
+      this.artifactInfo.sourceModelName,
+    );
+    this.artifactInfo.targetModelName = utils.toFileName(
+      options.destinationModel,
+    );
+    this.artifactInfo.targetModelPath = utils.pluralize(
+      this.artifactInfo.targetModelName,
+    );
+    this.artifactInfo.targetModelRequestBody = utils.camelCase(
+      this.artifactInfo.targetModelName,
+    );
+    this.artifactInfo.relationPropertyName = utils.pluralize(
```

Comment:
Sorry I realized that I made a mistake in my PR. And this line should be 
`this.artifactInfo.relationPropertyName = options.relationName;`

Reply:
My PR https://github.com/strongloop/loopback-next/pull/4515 that fixes it, might be a good reference for fixing it and adding tests.